File: src/2303.calculate-amount-paid-in-taxes.1382/solution.ts
function calculateTax (brackets: number[][], income: number): number {
    brackets.push([Infinity, brackets[brackets.length - 1][1], ]);
    let result = 0;
    let base = 0;
    for (const [upper, percent, ] of brackets) {
        if (income <= base) {
            break;
        }
        result += (Math.min(upper, income) - base) * percent / 100;
        base = upper;
    }

    return result;
}

File: src/2303.calculate-amount-paid-in-taxes.1382/solution.java
class Solution {
    public double calculateTax(int[][] brackets, int income) {
        double result = 0;
        int prev = 0;
        for (int[] bracket : brackets) {
            int diff = Math.min(income, bracket[0]) - prev;
            prev = bracket[0];
            result += diff * bracket[1] / 100.0;

            if (income <= bracket[0]) {
                break;
            }
        }
        return result;
    }
}